surl/__init__.py

```diff
@@ -227,30 +252,43 @@ def get_config_from_cli(parser, auth_dir):
                 "please delete it and login again:\n  $ rm {}".format(auth_path)
             )
 
-        return config, remainder
+        return args.url, config, remainder
+
+    store_env, store_type = get_environment_from_url(args.url)
+
+    if args.store:
+        store_env = args.store
+
+    if args.type:
+        store_type = args.type
+
+    if store_type == "snapcraft":
+        package_type = "snap"
+        default_permission = "package_access"
+    else:
+        package_type = "charm"
+        default_permission = "package-view"
 
-    # TODO: change this to use either snap or charm based on URL
     packages = (
-        [endpoints.Package(name, "snap") for name in args.snaps] if args.snaps else []
+        [endpoints.Package(name, package_type) for name in args.snaps] if args.snaps else []
```

Comment:
As mentioned above if more than 1 package is supported...maybe we could support adding the type as part of the name, e.g: `name:type` and fallback to charm if not specified (need to check if : is allowed in the snap names, probably not)

Reply:
Sounds good to me; I chose to fall back to "snap" instead of "charm" to be as close as possible to the old behaviour. Also changed the name of the option to "--package" to make it more generic.

Reply:
or it could be `type:name`...whatever you like more

Reply:
Oh yeah, I also did type:name without even thinking. :laughing: